app/charts/map/get-base-layer-style.ts

```diff
@@ -0,0 +1,5891 @@
+          "text-halo-width": 1,
+        },
+      },
+      {
+        id: "place_city",
+        type: "symbol",
+        source: "swissmaptiles",
+        "source-layer": "place",
+        minzoom: 2,
+        maxzoom: 14,
+        filter: ["all", ["==", "class", "city"]],
+        layout: {
+          "icon-image": [
+            "step",
+            ["zoom"],
+            "dot_circle_dark_grey_6",
+            6,
+            "dot_circle_dark_grey_8",
+            8,
+            "circle_circle_dark_grey_10",
+            10,
+            "circle_circle_dark_grey_12",
+            12,
+            "circle_circle_dark_grey_12",
+          ],
+          "icon-optional": false,
+          "icon-size": 1,
+          "symbol-sort-key": ["to-number", ["get", "rank"]],
+          "symbol-z-order": ["literal", "auto"],
+          "text-anchor": "bottom-left",
+          "text-field": translatedTextField,
+          "text-font": ["Frutiger Neue Condensed Bold"],
+          "text-justify": "auto",
+          "text-letter-spacing": 0.025,
+          "text-max-width": 10,
+          "text-offset": ["literal", [0.35, 0.1]],
+          "text-padding": 10,
+          "text-size": [
+            "interpolate",
+            ["cubic-bezier", 0.5, 0.1, 0.7, 1],
+            ["zoom"],
+            1,
+            11,
+            4,
+            12,
+            16,
+            48,
+          ],
+          "text-transform": "uppercase",
+          visibility: "visible",
+        },
+        paint: {
+          "icon-opacity": ["step", ["zoom"], 1, 11, 0],
+          "text-color": "rgba(64, 64, 64, 1)",
+          "text-halo-blur": 1,
+          "text-halo-color": "rgba(242, 242, 242, 0.6)",
+          "text-halo-width": [
+            "interpolate",
+            ["cubic-bezier", 0, 0.75, 0.25, 1],
+            ["zoom"],
+            12,
+            0,
+            16,
+            3,
+          ],
+          "text-opacity": 1,
+        },
+      },
+      {
+        id: "place_country",
+        type: "symbol",
+        source: "swissmaptiles",
+        "source-layer": "place",
+        minzoom: 0,
+        maxzoom: 7,
+        filter: ["all", ["in", "class", "country"], ["in", "iso_a2", "CH"]],
+        layout: {
+          "text-field": translatedTextField,
+          "text-font": ["Frutiger Neue Condensed Regular"],
+          "text-letter-spacing": 0.12,
+          "text-max-width": 10,
+          "text-size": ["interpolate", ["linear"], ["zoom"], 1, 12, 6, 16],
+          "text-transform": "uppercase",
+          visibility: "visible",
+        },
+        paint: {
+          "text-color": "rgba(145, 70, 145, 0.9)",
+          "text-halo-blur": 1,
+          "text-halo-color": "rgba(255, 255, 255, 0.8)",
+          "text-halo-width": 1,
+          "text-opacity": ["interpolate", ["linear"], ["zoom"], 6, 1, 7, 0],
+        },
+      },
+    ],
+    id: "b484873e-ac8c-45d6-a562-5402ab0f4985",
+  };
+};
```

Comment:
For me the question would be if we want to match Swisstopo's layout 1:1 at all times; what if they completely change their styling or add / remove some of the styling? Probably this shouldn't happen, but by having a static file with styles we would be safe that our styling won't change.

In fact, I was initially thinking that we could have our own styling (I think you also suggested this), so then it would make sense to have local files – right now they are the same as Swisstopo's styling, but if we'd like to change them, we would not depend on fetching Swisstopo's default styling and modifying it.

I also found it much easier to introduce some modifications to these layers when I could look at the styles in the editor, I think that if we wanted to modify some styles, there still would be a need to copy and paste their default JSON into text editor to exactly find what needs to be changed. But maybe it's only me, as I don't have a lot of experience with Mapbox styling 😅

What do you think?

Reply:
OK I see where you come from, let's see how it evolves in the future.